pkg/monitor/monitor/monitor_manager.go

```diff
@@ -252,6 +256,37 @@ func (mm *MonitorManager) syncTidbMonitorSecret(monitor *v1alpha1.TidbMonitor) (
 }
 
 func (mm *MonitorManager) syncTidbMonitorConfig(tc *v1alpha1.TidbCluster, monitor *v1alpha1.TidbMonitor) (*corev1.ConfigMap, error) {
+	if features.DefaultFeatureGate.Enabled(features.AutoScaling) {
+		// TODO: We need to update the status to tell users we are monitoring extra clusters
+		// Get all autoscaling clusters for TC, and add them to .Spec.Clusters to
+		// generate Prometheus config without modifying the original TidbMonitor
+		cloned := monitor.DeepCopy()
+		for _, tcRef := range monitor.Spec.Clusters {
+			r1, err := labels.NewRequirement(label.AutoInstanceLabelKey, selection.Exists, nil)
+			if err != nil {
+				klog.Errorf("tm[%s/%s] gets tc[%s/%s]'s autoscaling clusters failed, err: %v", monitor.Namespace, monitor.Name, tcRef.Namespace, tcRef.Name, err)
+				continue
+			}
+			r2, err := labels.NewRequirement(label.BaseTCLabelKey, selection.Equals, []string{tcRef.Name})
+			if err != nil {
+				klog.Errorf("tm[%s/%s] gets tc[%s/%s]'s autoscaling clusters failed, err: %v", monitor.Namespace, monitor.Name, tcRef.Namespace, tcRef.Name, err)
+				continue
+			}
+			selector := labels.NewSelector().Add(*r1).Add(*r2)
+			tcList, err := mm.cli.PingcapV1alpha1().TidbClusters(tcRef.Namespace).List(metav1.ListOptions{LabelSelector: selector.String()})
+			if err != nil {
+				klog.Errorf("tm[%s/%s] gets tc[%s/%s]'s autoscaling clusters failed, err: %v", monitor.Namespace, monitor.Name, tcRef.Namespace, tcRef.Name, err)
+				continue
+			}
+			for _, autoTc := range tcList.Items {
```

Comment:
If there are many TCs and the order of different items from the TC listers varies each time, this may cause the content of configmap changes each time, which will cause unexpected reload of the Prometheus configuration.